lib/cpfcnpj.ex

```diff
@@ -34,33 +34,32 @@ defmodule Cpfcnpj do
   """
   @spec valid?({:cpf | :cnpj, String.t()}) :: boolean()
   def valid?(number_in) do
-    if check_number(number_in) != :error, do: type_checker(number_in), else: false
+    check_number(number_in) and type_checker(number_in) and special_checker(number_in)
   end
 
   defp check_number({_, nil}) do
-    :error
+    false
   end
 
+  # Checks length and if all are equal
   defp check_number(tp_cpfcnpj) do
     cpfcnpj = String.replace(elem(tp_cpfcnpj, 1), ~r/[\.\/-]/, "")
 
     all_equal =
-      String.replace(cpfcnpj, String.at(cpfcnpj, 0), "")
+      cpfcnpj
+      |> String.replace(String.at(cpfcnpj, 0), "")
       |> String.length()
```

Comment:
O que acha de fazer algo assim? Fica mais expressivo.
```suggestion
    all_equal? =
      cpfcnpj
      |> String.replace(String.at(cpfcnpj, 0), "")
      |> String.length() == 0
```